src/main/scala/package.scala

```diff
@@ -284,6 +286,41 @@ sealed trait BSONFormats extends LowerImplicitBSONHandlers {
     }
   }
 
+  implicit object BSONDecimalFormat extends PartialFormat[BSONDecimal] {
+
+    val partialReads: PartialFunction[JsValue, JsResult[BSONDecimal]] = {
+      case DecimalRead(v) => JsSuccess(BSONDecimal(v._1, v._2))
+    }
+
+    val partialWrites: PartialFunction[BSONValue, JsValue] = {
+      @SuppressWarnings(Array("LooksLikeInterpolatedString"))
+      @inline def json: PartialFunction[BSONValue, JsValue] = {
+        case DecimalWrite(v) => v
+      }
+
+      json
+    }
+
+    private[json] object DecimalRead {
+      @SuppressWarnings(Array("LooksLikeInterpolatedString"))
+      def unapply(obj: JsObject): Option[BSONDecimal] =
+        (obj \ f"$$decimal").asOpt[JsValue].flatMap(strict) orElse
+          (obj \ f"$$decimal" \ f"$$numberDecimal").asOpt[JsValue].flatMap(strict)
+    }
+
+    private val strict: PartialFunction[JsValue, Option[BSONDecimal]] = {
+      case JsNumber(n) => BSONDecimal.fromBigDecimal(n).toOption
+      case JsString(n) => Try(BigDecimal(n)).map(BSONDecimal.fromBigDecimal).flatten.toOption
```

Comment:
`aTry.map(inner => anotherTry).flatten` is a code smell ~> = `.flatMap`

```suggestion
      case JsString(n) => Try(BigDecimal(n)).flatMap(BSONDecimal.fromBigDecimal).toOption
```

Reply:
Thx